lib/helpers/ScriptApiHelper.ts

```diff
@@ -8,45 +8,59 @@ export class ScriptApiHelper {
   }
 
   async get(path: string) {
-    const response = await this.api.get(this.api.baseUrl + '/umbraco/management/api/v1/script?path=' + path);
-    const json = await response.json();
-    if (json !== null) {
-      return json;
-    }
-    return null;
+    const response = await this.api.get(this.api.baseUrl + '/umbraco/management/api/v1/script/' + encodeURIComponent(path));
+    return await response.json();
   }
 
   async doesExist(path: string) {
-    const response = await this.api.get(this.api.baseUrl + '/umbraco/management/api/v1/script?path=' + path);
+    const response = await this.api.get(this.api.baseUrl + '/umbraco/management/api/v1/script/' + encodeURIComponent(path));
     return response.status() === 200;
   }
 
-  async create(name: string, content: string, parentPath: string = "") {
+  async create(name: string, content: string, parentPath: string = "/") {
     const scriptData = {
       "name": name,
-      "content": content,
-      "parentPath": parentPath
+      "parent": {
+        "path": parentPath
+      },
+      "content": content
     };
     const response = await this.api.post(this.api.baseUrl + '/umbraco/management/api/v1/script', scriptData);
     // Returns the path of the created Script
-    return response.headers().location.split("=").pop();
+    const path = response.headers().location.split("/v1/script/").pop();
+
+    if (path !== undefined) {
```

Comment:
In order to handle the "undefined" scenario, you should check whether the response itself is undefined, rather than checking the path.